pandas/core/frame.py

```diff
@@ -4635,7 +4636,16 @@ def quantile(self, q=0.5, axis=0, numeric_only=True):
             0 <= q <= 1, the quantile(s) to compute
         axis : {0, 1, 'index', 'columns'} (default 0)
             0 or 'index' for row-wise, 1 or 'columns' for column-wise
```

Comment:
i would like you move the doc-string to ``core/generic.py`` (and just show a ``NotImplementedMethod``), kind of like how we did for ``.reindex_axis`` (e.g. you setup ``_shared_docs``)